src/main/java/datastructs/ISample.java

```diff
@@ -9,32 +9,33 @@
     /**
      * The name of the sample
      */
-    public String getName();
+    String name();
 
     /**
      * Returns the size of the sample
      */
-    public int getsize();
+    int size();
 
 	/**
 	 * Add the value to the sample
 	 */
-	public void add(T value);
+	void add(T value);
```

Comment:
why the T? it's because this is an interface and you want to have different implementations down the road?

Reply:
It's not about the implementations is about the data types we want to store. ```NumericSample``` only uses ```Double``` whilst ```CategoricalSample``` only strings. This way we have a uniform interface